xml/src/main/scala/org/broadinstitute/monster/extractors/xml/XmlExtractor.scala

```diff
@@ -89,37 +141,66 @@ class XmlExtractor[Path] private[xml] (
       })
     }
 
-    groupXmlTags(xmlEventStream, Chain.empty, inTag = false).stream
-      .chunkN(tagsPerFile)
-      .evalMap { xmlChunk =>
-        IO.delay(File.newTemporaryFile()).flatMap { file =>
-          Resource.make {
-            IO.delay(file.newOutputStream)
-          } { outputStream =>
-            IO.delay(outputStream.close())
-          }.use { outputStream =>
-            xmlChunk.traverse { xmlEvents =>
-              Resource.make {
-                IO.delay {
-                  new JsonXMLOutputFactory(jsonXMLConfig)
-                    .createXMLEventWriter(outputStream)
-                }
-              } { writer =>
-                IO.delay {
-                  writer.close()
-                }
-              }.use { writer =>
-                xmlEvents.traverse_ { xmlEvent =>
-                  IO.delay {
-                    writer.add(xmlEvent)
-                  }
-                }
-              }.flatMap { _ =>
-                IO.delay(outputStream.write('\n'))
+    // Pull the stream until we find the first start element, which is the document root.
+    xmlEventStream
+      .dropWhile(!_.isStartElement)
+      .pull
+      .uncons1
+      .flatMap {
+        // No root found (document is empty?)
+        case None => Pull.done
```

Comment:
Do we have a test for this?

Reply:
We do now 😎 Nice catch, there was a bug.